mesheryctl/internal/cli/root/system/start.go

```diff
@@ -341,14 +341,33 @@ func start() error {
 		// downloaded required files successfully now apply the manifest files
 		log.Info("Starting Meshery...")
 
+		spinner := utils.CreateDefaultSpinner("Deploying Meshery on Kubernetes.", "...Meshery deployed on Kubernetes.")
+		spinner.Start()
+
 		// apply the adapters mentioned in the config.yaml file to the Kubernetes cluster
 		err = utils.ApplyManifestFiles(manifests, RequestedAdapters, kubeClient, false, false)
-
 		if err != nil {
-			return err
+			break
 		}
 
-		log.Info("...Meshery deployed on Kubernetes.")
+		deadline := time.Now().Add(10 * time.Second)
```

Comment:
It might take longer than 10 sec on first install. @leecalcote @hexxdump what should we do here?

Reply:
What’s the behavior after it hits the deadline? Does it back off and check every three seconds?

Reply:
No, it will basically skip the checking and proceed to say "Meshery is started."

We can let it check indefinitely but if there are any issues with the deployment then it will keep checking. Hence we have a timeout here. What should be the ideal "time out" time is the question here.

Reply:
As @leecalcote mentioned, we can not wait forever, there has to be an ultimate deadline.
Here are my thoughts: After the ultimate deadline is met, based on return value of utils.IsMesheryRunning(), we modify the FinalMsg to either “All pods started” or “Pod creation in progress, check kubectl get pods -n meshery before using Meshery”. I think we could modify IsMesheryRunning() or write a new utility to report the status of Pod/Deployment, rather than just checking for whether the deployment is listed or not(it could very well be a deployment in progress). Also, it would help, if we could actually display the status of deployment on the ultimate deadline.